There's a build issue with the frontend:

```console
 nix-build -A open-webui.frontend
this derivation will be built:
  /nix/store/0db1cjs58wazkjdklginca45ac0klmbh-open-webui-frontend-0.6.17.drv
building '/nix/store/0db1cjs58wazkjdklginca45ac0klmbh-open-webui-frontend-0.6.17.drv'...
Running phase: unpackPhase
unpacking source archive /nix/store/wlyqbdgg81f9mzl43rzcarzn8pznj2s3-source
source root is source
Running phase: patchPhase
Executing npmConfigHook
Configuring npm
Validating consistency between /build/source/package-lock.json and /nix/store/4k790y1qp7bxq3jvqj12l830qg0jzh6g-open-webui-frontend-0.6.17-npm-deps/package-lock.json
Setting npm_config_cache to /nix/store/4k790y1qp7bxq3jvqj12l830qg0jzh6g-open-webui-frontend-0.6.17-npm-deps
Installing dependencies
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @tiptap/extension-bubble-menu@2.26.1
npm warn Found: @tiptap/core@3.0.7
npm warn node_modules/@tiptap/core
npm warn   @tiptap/core@"^3.0.7" from the root project
npm warn   28 more (@tiptap/extension-blockquote, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer @tiptap/core@"^2.7.0" from @tiptap/extension-bubble-menu@2.26.1
npm warn node_modules/@tiptap/extension-bubble-menu
npm warn   @tiptap/extension-bubble-menu@"^2.26.1" from the root project
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @tiptap/extension-bubble-menu@2.26.1
npm warn Found: @tiptap/core@3.0.7
npm warn node_modules/@tiptap/core
npm warn   @tiptap/core@"^3.0.7" from the root project
npm warn   28 more (@tiptap/extension-blockquote, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer @tiptap/core@"^3.0.7" from @tiptap/extension-code-block@3.0.7
npm warn node_modules/@tiptap/extension-code-block
npm warn   peer @tiptap/extension-code-block@"^3.0.7" from @tiptap/extension-code-block-lowlight@3.0.7
npm warn   node_modules/@tiptap/extension-code-block-lowlight
npm warn   1 more (@tiptap/starter-kit)
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @tiptap/extension-bubble-menu@2.26.1
npm warn Found: @tiptap/core@3.0.7
npm warn node_modules/@tiptap/core
npm warn   @tiptap/core@"^3.0.7" from the root project
npm warn   28 more (@tiptap/extension-blockquote, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer @tiptap/core@"^3.0.7" from @tiptap/extension-code-block-lowlight@3.0.7
npm warn node_modules/@tiptap/extension-code-block-lowlight
npm warn   @tiptap/extension-code-block-lowlight@"^3.0.7" from the root project
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @tiptap/extension-bubble-menu@2.26.1
npm warn Found: @tiptap/core@3.0.7
npm warn node_modules/@tiptap/core
npm warn   @tiptap/core@"^3.0.7" from the root project
npm warn   28 more (@tiptap/extension-blockquote, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer @tiptap/core@"^3.0.7" from @tiptap/extension-collaboration@3.0.7
npm warn node_modules/@tiptap/extension-collaboration
npm warn   peer @tiptap/extension-collaboration@"^3.0.7" from @tiptap/extension-drag-handle@3.0.7
npm warn   node_modules/@tiptap/extension-drag-handle
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @tiptap/extension-bubble-menu@2.26.1
npm warn Found: @tiptap/core@3.0.7
npm warn node_modules/@tiptap/core
npm warn   @tiptap/core@"^3.0.7" from the root project
npm warn   28 more (@tiptap/extension-blockquote, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer @tiptap/core@"^3.0.7" from @tiptap/extension-drag-handle@3.0.7
npm warn node_modules/@tiptap/extension-drag-handle
npm warn   @tiptap/extension-drag-handle@"^3.0.7" from the root project
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @tiptap/extension-bubble-menu@2.26.1
npm warn Found: @tiptap/core@3.0.7
npm warn node_modules/@tiptap/core
npm warn   @tiptap/core@"^3.0.7" from the root project
npm warn   28 more (@tiptap/extension-blockquote, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer @tiptap/core@"^3.0.7" from @tiptap/extension-file-handler@3.0.7
npm warn node_modules/@tiptap/extension-file-handler
npm warn   @tiptap/extension-file-handler@"^3.0.7" from the root project
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @tiptap/extension-bubble-menu@2.26.1
npm warn Found: @tiptap/core@3.0.7
npm warn node_modules/@tiptap/core
npm warn   @tiptap/core@"^3.0.7" from the root project
npm warn   28 more (@tiptap/extension-blockquote, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer @tiptap/core@"^2.7.0" from @tiptap/extension-floating-menu@2.26.1
npm warn node_modules/@tiptap/extension-floating-menu
npm warn   @tiptap/extension-floating-menu@"^2.26.1" from the root project
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @tiptap/extension-bubble-menu@2.26.1
npm warn Found: @tiptap/core@3.0.7
npm warn node_modules/@tiptap/core
npm warn   @tiptap/core@"^3.0.7" from the root project
npm warn   28 more (@tiptap/extension-blockquote, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer @tiptap/core@"^3.0.7" from @tiptap/extension-highlight@3.0.7
npm warn node_modules/@tiptap/extension-highlight
npm warn   @tiptap/extension-highlight@"^3.0.7" from the root project
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @tiptap/extension-bubble-menu@2.26.1
npm warn Found: @tiptap/core@3.0.7
npm warn node_modules/@tiptap/core
npm warn   @tiptap/core@"^3.0.7" from the root project
npm warn   28 more (@tiptap/extension-blockquote, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer @tiptap/core@"^3.0.7" from @tiptap/extension-image@3.0.7
npm warn node_modules/@tiptap/extension-image
npm warn   @tiptap/extension-image@"^3.0.7" from the root project
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @tiptap/extension-bubble-menu@2.26.1
npm warn Found: @tiptap/core@3.0.7
npm warn node_modules/@tiptap/core
npm warn   @tiptap/core@"^3.0.7" from the root project
npm warn   28 more (@tiptap/extension-blockquote, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer @tiptap/core@"^3.0.7" from @tiptap/extension-link@3.0.7
npm warn node_modules/@tiptap/extension-link
npm warn   @tiptap/extension-link@"^3.0.7" from the root project
npm warn   1 more (@tiptap/starter-kit)
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @tiptap/extension-bubble-menu@2.26.1
npm warn Found: @tiptap/core@3.0.7
npm warn node_modules/@tiptap/core
npm warn   @tiptap/core@"^3.0.7" from the root project
npm warn   28 more (@tiptap/extension-blockquote, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer @tiptap/core@"^3.0.7" from @tiptap/extension-list@3.0.7
npm warn node_modules/@tiptap/extension-list
npm warn   @tiptap/extension-list@"^3.0.7" from the root project
npm warn   5 more (@tiptap/extension-bullet-list, ...)
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @tiptap/extension-bubble-menu@2.26.1
npm warn Found: @tiptap/core@3.0.7
npm warn node_modules/@tiptap/core
npm warn   @tiptap/core@"^3.0.7" from the root project
npm warn   28 more (@tiptap/extension-blockquote, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer @tiptap/core@"^3.0.7" from @tiptap/extension-node-range@3.0.7
npm warn node_modules/@tiptap/extension-node-range
npm warn   peer @tiptap/extension-node-range@"^3.0.7" from @tiptap/extension-drag-handle@3.0.7
npm warn   node_modules/@tiptap/extension-drag-handle
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @tiptap/extension-bubble-menu@2.26.1
npm warn Found: @tiptap/core@3.0.7
npm warn node_modules/@tiptap/core
npm warn   @tiptap/core@"^3.0.7" from the root project
npm warn   28 more (@tiptap/extension-blockquote, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer @tiptap/core@"^3.0.7" from @tiptap/extension-table@3.0.7
npm warn node_modules/@tiptap/extension-table
npm warn   @tiptap/extension-table@"^3.0.7" from the root project
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @tiptap/extension-bubble-menu@2.26.1
npm warn Found: @tiptap/core@3.0.7
npm warn node_modules/@tiptap/core
npm warn   @tiptap/core@"^3.0.7" from the root project
npm warn   28 more (@tiptap/extension-blockquote, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer @tiptap/core@"^3.0.7" from @tiptap/extension-text-style@3.0.7
npm warn node_modules/@tiptap/extension-text-style
npm warn   peer @tiptap/extension-text-style@"^3.0.7" from @tiptap/extension-file-handler@3.0.7
npm warn   node_modules/@tiptap/extension-file-handler
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @tiptap/extension-bubble-menu@2.26.1
npm warn Found: @tiptap/core@3.0.7
npm warn node_modules/@tiptap/core
npm warn   @tiptap/core@"^3.0.7" from the root project
npm warn   28 more (@tiptap/extension-blockquote, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer @tiptap/core@"^3.0.7" from @tiptap/extension-typography@3.0.7
npm warn node_modules/@tiptap/extension-typography
npm warn   @tiptap/extension-typography@"^3.0.7" from the root project
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @tiptap/extension-bubble-menu@2.26.1
npm warn Found: @tiptap/core@3.0.7
npm warn node_modules/@tiptap/core
npm warn   @tiptap/core@"^3.0.7" from the root project
npm warn   28 more (@tiptap/extension-blockquote, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer @tiptap/core@"^3.0.7" from @tiptap/extension-youtube@3.0.7
npm warn node_modules/@tiptap/extension-youtube
npm warn   @tiptap/extension-youtube@"^3.0.7" from the root project
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @tiptap/extension-bubble-menu@2.26.1
npm warn Found: @tiptap/core@3.0.7
npm warn node_modules/@tiptap/core
npm warn   @tiptap/core@"^3.0.7" from the root project
npm warn   28 more (@tiptap/extension-blockquote, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer @tiptap/core@"^3.0.7" from @tiptap/extensions@3.0.7
npm warn node_modules/@tiptap/extensions
npm warn   @tiptap/extensions@"^3.0.7" from the root project
npm warn   3 more (@tiptap/extension-dropcursor, ...)
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @tiptap/extension-bubble-menu@2.26.1
npm warn Found: @tiptap/core@undefined
npm warn node_modules/@tiptap/core
npm warn   @tiptap/core@"^3.0.7" from the root project
npm warn
npm warn Could not resolve dependency:
npm warn peer @tiptap/core@"^2.7.0" from @tiptap/extension-bubble-menu@2.26.1
npm warn node_modules/@tiptap/extension-bubble-menu
npm warn   @tiptap/extension-bubble-menu@"^2.26.1" from the root project
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @tiptap/extension-bubble-menu@2.26.1
npm warn Found: @tiptap/pm@3.0.7
npm warn node_modules/@tiptap/pm
npm warn   @tiptap/pm@"^3.0.7" from the root project
npm warn   7 more (@tiptap/extension-code-block-lowlight, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer @tiptap/pm@"^2.7.0" from @tiptap/extension-bubble-menu@2.26.1
npm warn node_modules/@tiptap/extension-bubble-menu
npm warn   @tiptap/extension-bubble-menu@"^2.26.1" from the root project
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @tiptap/extension-bubble-menu@2.26.1
npm warn Found: @tiptap/pm@3.0.7
npm warn node_modules/@tiptap/pm
npm warn   @tiptap/pm@"^3.0.7" from the root project
npm warn   7 more (@tiptap/extension-code-block-lowlight, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer @tiptap/pm@"^3.0.7" from @tiptap/extension-code-block-lowlight@3.0.7
npm warn node_modules/@tiptap/extension-code-block-lowlight
npm warn   @tiptap/extension-code-block-lowlight@"^3.0.7" from the root project
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @tiptap/extension-bubble-menu@2.26.1
npm warn Found: @tiptap/pm@3.0.7
npm warn node_modules/@tiptap/pm
npm warn   @tiptap/pm@"^3.0.7" from the root project
npm warn   7 more (@tiptap/extension-code-block-lowlight, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer @tiptap/pm@"^3.0.7" from @tiptap/extension-collaboration@3.0.7
npm warn node_modules/@tiptap/extension-collaboration
npm warn   peer @tiptap/extension-collaboration@"^3.0.7" from @tiptap/extension-drag-handle@3.0.7
npm warn   node_modules/@tiptap/extension-drag-handle
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @tiptap/extension-bubble-menu@2.26.1
npm warn Found: @tiptap/pm@3.0.7
npm warn node_modules/@tiptap/pm
npm warn   @tiptap/pm@"^3.0.7" from the root project
npm warn   7 more (@tiptap/extension-code-block-lowlight, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer @tiptap/pm@"^3.0.7" from @tiptap/extension-drag-handle@3.0.7
npm warn node_modules/@tiptap/extension-drag-handle
npm warn   @tiptap/extension-drag-handle@"^3.0.7" from the root project
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @tiptap/extension-bubble-menu@2.26.1
npm warn Found: @tiptap/pm@3.0.7
npm warn node_modules/@tiptap/pm
npm warn   @tiptap/pm@"^3.0.7" from the root project
npm warn   7 more (@tiptap/extension-code-block-lowlight, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer @tiptap/pm@"^3.0.7" from @tiptap/extension-file-handler@3.0.7
npm warn node_modules/@tiptap/extension-file-handler
npm warn   @tiptap/extension-file-handler@"^3.0.7" from the root project
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @tiptap/extension-bubble-menu@2.26.1
npm warn Found: @tiptap/pm@3.0.7
npm warn node_modules/@tiptap/pm
npm warn   @tiptap/pm@"^3.0.7" from the root project
npm warn   7 more (@tiptap/extension-code-block-lowlight, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer @tiptap/pm@"^2.7.0" from @tiptap/extension-floating-menu@2.26.1
npm warn node_modules/@tiptap/extension-floating-menu
npm warn   @tiptap/extension-floating-menu@"^2.26.1" from the root project
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @tiptap/extension-bubble-menu@2.26.1
npm warn Found: @tiptap/pm@3.0.7
npm warn node_modules/@tiptap/pm
npm warn   @tiptap/pm@"^3.0.7" from the root project
npm warn   7 more (@tiptap/extension-code-block-lowlight, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer @tiptap/pm@"^3.0.7" from @tiptap/extension-node-range@3.0.7
npm warn node_modules/@tiptap/extension-node-range
npm warn   peer @tiptap/extension-node-range@"^3.0.7" from @tiptap/extension-drag-handle@3.0.7
npm warn   node_modules/@tiptap/extension-drag-handle
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @tiptap/extension-bubble-menu@2.26.1
npm warn Found: @tiptap/pm@3.0.7
npm warn node_modules/@tiptap/pm
npm warn   @tiptap/pm@"^3.0.7" from the root project
npm warn   7 more (@tiptap/extension-code-block-lowlight, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer @tiptap/pm@"^3.0.7" from @tiptap/extension-table@3.0.7
npm warn node_modules/@tiptap/extension-table
npm warn   @tiptap/extension-table@"^3.0.7" from the root project
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @tiptap/extension-bubble-menu@2.26.1
npm warn Found: @tiptap/pm@undefined
npm warn node_modules/@tiptap/pm
npm warn   @tiptap/pm@"^3.0.7" from the root project
npm warn
npm warn Could not resolve dependency:
npm warn peer @tiptap/pm@"^2.7.0" from @tiptap/extension-bubble-menu@2.26.1
npm warn node_modules/@tiptap/extension-bubble-menu
npm warn   @tiptap/extension-bubble-menu@"^2.26.1" from the root project
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @tiptap/extension-code-block-lowlight@3.0.7
npm warn Found: @tiptap/extension-code-block@undefined
npm warn node_modules/@tiptap/extension-code-block
npm warn
npm warn Could not resolve dependency:
npm warn peer @tiptap/extension-code-block@"^3.0.7" from @tiptap/extension-code-block-lowlight@3.0.7
npm warn node_modules/@tiptap/extension-code-block-lowlight
npm warn   @tiptap/extension-code-block-lowlight@"^3.0.7" from the root project
npm error code ENOTCACHED
npm error request to https://registry.npmjs.org/@tiptap%2fcore failed: cache mode is 'only-if-cached' but no cached response is available.
npm error Log files were not written due to an error writing to the directory: /nix/store/4k790y1qp7bxq3jvqj12l830qg0jzh6g-open-webui-frontend-0.6.17-npm-deps/_logs
npm error You can rerun the command with `--loglevel=verbose` to see the logs in your terminal

ERROR: npm failed to install dependencies

Here are a few things you can try, depending on the error:
1. Set `makeCacheWritable = true`
  Note that this won't help if npm is complaining about not being able to write to the logs directory -- look above that for the actual error.
2. Set `npmFlags = [ "--legacy-peer-deps" ]`
```

--- a/pkgs/by-name/op/open-webui/package.nix
+++ b/pkgs/by-name/op/open-webui/package.nix
@@ -9,13 +9,13 @@
 }:
 let
   pname = "open-webui";
-  version = "0.6.16";
+  version = "0.6.18";
 
   src = fetchFromGitHub {
     owner = "open-webui";
     repo = "open-webui";
     tag = "v${version}";
-    hash = "sha256-git5bNvW+CfeH4N9/AF5iWly0KoJf6OZKDKlGfhDtP4=";
+    hash = "sha256-1V9mOhO8jpr0HU0djLjKw6xDQMBmqie6Gte4xfg9PfQ=";
   };
 
   frontend = buildNpmPackage rec {
@@ -26,13 +26,19 @@ let
     # must match lock file in open-webui
     # TODO: should we automate this?
     # TODO: with JQ? "jq -r '.packages["node_modules/pyodide"].version' package-lock.json"
-    pyodideVersion = "0.27.3";
+    pyodideVersion = "0.28.0";
     pyodide = fetchurl {
-      hash = "sha256-SeK3RKqqxxLLf9DN5xXuPw6ZPblE6OX9VRXMzdrmTV4=";
+      hash = "sha256-4YwDuhcWPYm40VKfOEqPeUSIRQl1DDAdXEUcMuzzU7o=";
       url = "https://github.com/pyodide/pyodide/releases/download/${pyodideVersion}/pyodide-${pyodideVersion}.tar.bz2";
     };
 
-    npmDepsHash = "sha256-J739sG2rKmgQQ2bPHtwzdmYcwCig3AMzpHxt+PhmAME=";
+    npmDepsHash = "sha256-bMqK9NvuTwqnhflGDfZTEkaFG8y34Qf94SgR0HMClrQ=";
+
+    # See https://github.com/open-webui/open-webui/issues/15880
+    npmFlags = [
+      "--force"
+      "--legacy-peer-deps"
+    ];
 
     # Disabling `pyodide:fetch` as it downloads packages during `buildPhase`
     # Until this is solved, running python packages from the browser will not work.
@@ -163,6 +169,7 @@ python3Packages.buildPythonApplication rec {
       pillow
       pinecone-client
       playwright
+      posthog
       psutil
       psycopg2-binary
       pycrdt

